### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by Nathan24zz**
Initial commit

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+build
+log
+install
+
+__pycache__
+*.pyc
+
+*.cfg
+*.names
+*.weights
```
Comment (segara2410) [inline .gitignore]: Add a newline at the end of the file.

File: ninshiki_yolo/detector.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+import argparse
+import cv2
+import numpy as np
+import rclpy
+from rclpy.node import Node
+from shisen_interfaces.msg import Image
+
+
+class Detector (Node):
+    def __init__(self, node_name, topic_name, config, names, weights):
```
Comment (segara2410) [inline ninshiki_yolo/detector.py]: Add type hint for function parameters and return value

**Review by segara2410 — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -1,2 +1,11 @@
-# ninshiki_yolo
-ROS 2 object detection package with YOLO
+# Ninshiki YOLO
+
+[![latest version](https://img.shields.io/github/v/release/ichiro-its/ninshiki.svg)](https://github.com/ichiro-its/ninshiki/releases/)
+[![license](https://img.shields.io/github/license/ichiro-its/ninshiki.svg)](./LICENSE)
+[![build and test status](https://github.com/ichiro-its/ninshiki/actions/workflows/build-and-test.yml/badge.svg)](https://github.com/ichiro-its/ninshiki/actions)
```
Comment (threeal) [inline README.md]: The `build-and-test` workflow doesn't exist, yet you already add badge for it.

File: package.xml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>ninshiki_yolo</name>
+  <version>0.0.0</version>
```
Comment (threeal) [inline package.xml]: I actually prefer to start project's version from `0.1.0`, but it's actually fine to use that.

File: package.xml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  <description>An object detection package with YOLO</description>
+  <maintainer email="nathaterbaik@gmail.com">ichiro</maintainer>
+  <license>MIT License</license>
+  <exec_depend>shisen_interface</exec_depend>
+  <exec_depend>ninshiki_interface</exec_depend>
```
Comment (threeal) [inline package.xml]: what is `ninshiki_interface` used for?

**Review by threeal — CHANGES_REQUESTED** [review]
See individual comment for the requested changes.

**Review by threeal — APPROVED** [review]
Looks good to me.

**Review by segara2410 — APPROVED** [review]
lgtm

**Pull request merged**